### ISSUE TITLE
Refactor racer argument generation

### ISSUE THREAD

**Issue by jwilm** [jwilm/racerd#3]
There's a block of code copied [here](https://github.com/jwilm/racerd/blob/master/src/engine/racer.rs#L24-L36) and [here](https://github.com/jwilm/racerd/blob/master/src/engine/racer.rs#L68-L82). It would be nice to have a function like

```rust
fn prepare_racer_args(ctx: &Context) ->  (::racer::core::Session, usize, path)
```

which wraps what the linked code does. I failed at refactoring this the first time due to some lifetime errors I couldn't figure out. The calls to `session.cache_file_contents` somehow result in an error

> `Session` does not live long enough

**Comment by jwilm** [comment]
I found the exact compiler error in my logs. The method signature I was trying to make work can be seen as well.

```
   Compiling libracerd v0.1.0 (file:///Users/jwilm/code/racerd)
src/engine/racer.rs:17:9: 17:16 error: `session` does not live long enough
src/engine/racer.rs:17         session.cache_file_contents(buffer.path(), &buffer.contents[..]);
                               ^~~~~~~
src/engine/racer.rs:10:89: 27:2 note: reference must be valid for the lifetime 'a as defined on the block at 10:88...
src/engine/racer.rs:10 fn build_session<'a>(ctx: &'a Context) -> (&'a Path, ::racer::core::Session<'a>, usize) {
src/engine/racer.rs:11     let path = ctx.query_path();
src/engine/racer.rs:12
src/engine/racer.rs:13     let session = ::racer::core::Session::from_path(path, path);
src/engine/racer.rs:14
src/engine/racer.rs:15     // compiles successfully without this loop
```

The code that generated this error:

```rust
fn build_session<'a>(ctx: &'a Context) -> (&'a Path, ::racer::core::Session<'a>, usize) {
    let path = ctx.query_path();

    let session = ::racer::core::Session::from_path(path, path);

    // compiles successfully without this loop
    for buffer in ctx.buffers {
        session.cache_file_contents(buffer.path(), &buffer.contents[..]);
    }

    // let query_src = &session.load_file(path).src.code[..];
    // let pos = ::racer::scopes::coords_to_point(query_src,
    //                                            ctx.query_cursor.line,
    //                                            ctx.query_cursor.col);

    let pos = 0;
    (path, session, pos)
}
```

**Comment by jwilm** [comment]
This has been resolved as of #5 landing
